R/plot-qq.R

```diff
@@ -0,0 +1,108 @@
+#' @title plotQQ
+#' @description
+#' Producing Histograms
+#'
+#' @inheritParams addScatter
+#' @param dataMapping
+#' A `QQDataMapping` object mapping `y` and aesthetic groups to their variable names of `data`.
+#' @param plotConfiguration
+#' An optional `QQPlotConfiguration` object defining labels, grid, background and watermark.
+#' @return A `ggplot` object
+#'
+#' @export
+#' @family molecule plots
+#' @examples
+#' # Produce QQ plot of normally distributed data
+#' plotQQ(y = rnorm(100))
+#'
+#' # Produce QQ plot of normally distributed data split by group
+#' qqData <- data.frame(
+#'   residuals = c(rnorm(100), rnorm(100)),
+#'   groups = c(rep("Group A", 100), rep("Group B", 100))
+#' )
+#' plotQQ(
+#'   data = qqData,
+#'   dataMapping = QQDataMapping$new(y = "residuals", group = "groups")
+#' )
+#'
+plotQQ <- function(data = NULL,
+                   metaData = NULL,
+                   y = NULL,
+                   dataMapping = NULL,
+                   plotConfiguration = NULL,
+                   plotObject = NULL) {
+  #----- Validation and formatting of input arguments -----
+  if (is.null(data)) {
+    validateIsNumeric(y)
+    data <- data.frame(y = y)
+    dataMapping <- dataMapping %||% QQDataMapping$new(
+      y = ifNotNull(y, "y"),
+      data = data
+    )
+  }
```

Comment:
2 questions on this:
 - what is `y` and when is the function `plotQQ  `called with `data=NULL` and `y != NULL` ?
 - why do we need `ifNotNull `in the line 39? Isn't it assured that y is not null by `validateIsNumeric(y)` before?

Reply:
Using `data` and `dataMapping` are a bit heavy when most of cases are just displaying QQ plots for a vector.
For this reason, there is also an optional `y` input which is transformed internally into a `data` variable and its `dataMapping`

The examples below display the exact same plots
```r
qqData <- data.frame(y = rnorm(100))

plotQQ(y = qqData$y)

plotQQ(data = qqData, dataMapping = QQDataMapping$new(y="y"))
```

![image](https://user-images.githubusercontent.com/45343665/185586653-733c3271-f66d-435d-8ac8-c85a7b0d6b2f.png)

Reply:
nice!